src/index.js

```diff
@@ -27,11 +27,19 @@ function resolveExtends(extension, base) {
 	var result = merge(base);
 	for (let name in extension) {
 		let item = extension[name];
-		if (item.kind === 'interface' && !item.base) {
+		if (item.kind === 'interface' && name in base) {
 			let baseItem = base[name];
+			if (item.base)
+				for (const base of item.base)
+					if (baseItem.base.indexOf(base) === -1)
+						baseItem.base.push(base);
+
 			result[name] = merge(baseItem, {
 				props: merge(baseItem.props, item.props)
 			});
+		} else if (item.kind === 'enum' && name in base) {
+			result[name] = base[name];
+			result[name].values.push(...item.values);

```

Comment:
Hm, wait, doesn't this modify the base object? With `merge` used in other branches we create a copy, but this one seems to retain original object by reference.

Reply:
Oh, that's right — I didn't realize the base object is important, since I write the file immediately after the merge. I'll fix that.